### PR TITLE
Enable use of pre-commit for easy use in a git hook

### LINKED ISSUE #1541: Enable use of pre-commit for easy use in a git hook
**What version of this package are you using?**
latest

**What problem do you want to solve?**
Easier use of standard within a git hook.

**What do you think is the correct solution to this problem?**
Use of the [pre-commit](https://pre-commit.com/) library, like prettier does [here](https://prettier.io/docs/en/precommit.html#option-4-precise-commitshttpsgithubcomjameshenryprecise-commits)

**Are you willing to submit a pull request to implement this change?**
Yes, I have a fork already doing it (see the only addition [here](https://github.com/thclark/standard/blob/master/.pre-commit-hooks.yaml) and am happy to submit a PR if someone tells me what branch to make it into.

I'll also need to update the relevant section of the docs, but that's no problem to add, as long as this suggestion is welcome.

Using my fork of standard, my `.pre-commit-config.yaml` files simply contain:
```
  - repo: https://github.com/thclark/standard
    rev: master
    hooks:
      - id: standard
```

### PR DESCRIPTION
**Pull request by thclark**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Documentation update
[ ] Bug fix
[x] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
Added a `.pre-commit-hooks.yaml` file and updated README to give users instructions on how to install it.

**Which issue (if any) does this pull request address?**
Fixes #1541 

**Is there anything you'd like reviewers to focus on?**
n/a

### CODE REVIEW

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+- id: standard
+  name: standard
+  entry: standard --fix
+  language: node
+  files: "\\.(\
+    js|jsx\
+    |ts|tsx\
+    )$"
```
Comment (feross) [inline .pre-commit-hooks.yaml:8]: Can you also add support for .mjs and .cjs files?

File: README.md [hunk 2]
```diff
@@ -679,6 +684,38 @@ if [[ $? -ne 0 ]]; then
...
+```yaml
+  - repo: https://github.com/pre-commit/mirrors-eslint
+    rev: master
+    hooks:
+      - id: eslint
+        files: \.[jt]sx?$  # *.js, *.jsx, *.ts and *.tsx
+        types: [file]
+        additional_dependencies:
+          - eslint@latest
+          - eslint-plugin-standard@latest
```
Comment (feross) [inline README.md:709]: We don't use `eslint-plugin-standard` anymore. And I don't see `eslint-config-standard` included here.

I'd rather keep this section simple, tbh, and just remove the instructions for using a custom eslint config.

**Review by feross — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (thclark) [reply README.md]: @feross that explains a few things about my configuration, and to my mind serves as evidence that it actually should be in there. The unbelievable agony of figuring all this stuff myself would've been quickly bypassed by this useful snippet!

However, hearing your point I've made a second attempt, corrected plugin-->config and removed all the non-standard-related plugins (a helpful code snippet but perhaps not for here.

What do you think? If you still don't like it I'll remove it. But I honestly feel like that would waste much of the effort I went to!

Comment (feross) [reply README.md]: Looks good now

**Review by feross — APPROVED** [review]

**Pull request merged**